proxy/http2/Http2DependencyTree.h

```diff
@@ -37,80 +37,89 @@
 const static uint32_t K                               = 256;
 const static uint32_t HTTP2_DEPENDENCY_TREE_MAX_DEPTH = 256;
 
-template <typename T> class Http2DependencyTree
+namespace Http2DependencyTree
+{
+class Node
 {
 public:
-  class Node
+  Node(void *t = nullptr)
+    : active(false),
```

Comment:
That's not on our coding style and the C++11 guideline is provided for REVIEW :)
https://cwiki.apache.org/confluence/display/TS/Coding+Style

Joking aside, we need a checklist, and notes. The change should be made with a separate commit. (Including it on the same PR is fine with me.)